--- evas-1.1.0/src/lib/engines/common/evas_font_ot.c.orig	2011-10-18 15:17:19.000000000 +0200
+++ evas-1.1.0/src/lib/engines/common/evas_font_ot.c	2011-12-11 14:07:19.125506961 +0100
@@ -286,10 +286,10 @@
         slen = len;
      }
 
-   buffer = hb_buffer_create();
+   buffer = hb_buffer_create(0);
    hb_buffer_set_unicode_funcs(buffer, _evas_common_font_ot_unicode_funcs_get());
    hb_buffer_set_language(buffer, hb_language_from_string(
-            evas_common_language_from_locale_get(), -1));
+            evas_common_language_from_locale_get()));
    hb_buffer_set_script(buffer, _evas_script_to_harfbuzz[props->script]);
    hb_buffer_set_direction(buffer,
          (props->bidi.dir == EVAS_BIDI_DIRECTION_RTL) ?
